main.js

```diff
@@ -37,24 +37,13 @@ document.querySelectorAll(".number").forEach((number) => {
 document.querySelectorAll(".operator").forEach((operator) => {
   operator.addEventListener("click", (event) => {
     str += event.target.innerHTML;
-    //console.log(event.target.value)
+    const secondLastChar = str[str.length - 2];
+    // check if the last character is an operator
+    if (secondLastChar === "+" || secondLastChar === "-" || secondLastChar === "*" || secondLastChar === "/") {
+      str = str.slice(0, -2);
+      str += event.target.innerHTML;
+    }
+    
```

Comment:
To avoid multiple operand insertion, check to ensure that the second last operator is an operand, and if so, any operand that follows an operand is sliced off the input string.